File: Array/1816. Truncate Sentence/TruncateSentence.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    string truncateSentence(string s, int k) {
        string result = "";
        for (char ch : s) {
            if (ch == ' ') {
                k--;
            }
            if (k == 0) {
                break;
            }
            result += ch;
        }
        return result;
    }
};

int main() {
    Solution sol;
    string str = "Hello how are you Contestant";
    int k = 4;
    string result = sol.truncateSentence(str, k);
    cout << "Truncated sentence: " << result << endl;
    return 0;
}

File: Array/1816. Truncate Sentence/TruncateSentence.py
class Solution:
    def truncateSentence(self, s: str, k: int) -> str:
        result = ""
        for ch in s:
            if ch == ' ':
                k -= 1
            if k == 0:
                break
            result += ch
        return result

# 测试代码
if __name__ == "__main__":
    sol = Solution()
    s = "Hello how are you Contestant"
    k = 4
    result = sol.truncateSentence(s, k)
    print("Truncated sentence:", result)